repose-aggregator/components/filters/rate-limiting-filter/src/main/java/org/openrepose/filters/ratelimiting/RateLimitingHandler.java

```diff
@@ -109,9 +108,30 @@ private boolean requestHasExpectedHeaders(HttpServletRequest request) {
         return request.getHeader(PowerApiHeader.USER) != null;
     }
 
+    private void setRequestedMediaType(List<String> acceptValues) {
+        if (acceptValues.isEmpty()) {
+            originalPreferredAccept = DEFAULT_MIME_TYPE;
+        } else {
+            originalPreferredAccept = acceptValues.stream()
+                    .map(HeaderValue::new)
+                    .filter(headerValue -> headerValue.quality() != 0.0)
+                    .sorted(Comparator.comparingDouble(HeaderValue::quality).reversed())
+                    .map(HeaderValue::value)
+                    .map(MimeType::getBestFitMimeType)
+                    .filter(mimeType -> SUPPORTED_MIME_TYPES.stream().anyMatch(mimeType::matches))
+                    .findFirst()
+                    .orElse(MimeType.UNKNOWN);
+        }
+    }
+
     private FilterAction describeLimitsForRequest(HttpServletRequestWrapper request, HttpServletResponseWrapper response) {
         if (originalPreferredAccept == MimeType.UNKNOWN) {
+            // If the accept header is set to a media type that the rate limiting filter cannot handle,
+            // the rate limiting filter should return a 406, or a body with some default media type
+            // (ignoring the accept header). HTTP/1.0 spec dictates that the former approach be taken,
+            // while HTTP/1.1 leaves it as an implementation decision. We chose to comply with the former.
             response.setStatus(SC_NOT_ACCEPTABLE);
+            // TODO: Set the supported accept types in the body? Provide a link to a supported types page?
```

Comment:
I wasn't sure what we wanted to do here. There does not seem to be specified (by RFC) behavior for automatic content-negotiation with regards to accept type. It is suggested, however, that the response payload contain the supported media types. Of course, the format is not specified, so... 

> The server SHOULD generate a payload containing a list of available
>  representation characteristics and corresponding resource identifiers
> from which the user or user agent can choose the one most
> appropriate.  A user agent MAY automatically select the most
>  appropriate choice from that list.  However, this specification does
> not define any standard for such automatic selection, as described in
> Section 6.4.1.

Reference: https://tools.ietf.org/html/rfc7231#section-6.5.6